### DISCUSSION TITLE
Hostonly usage for Hyper-V and WSL Running on Windows 10 22 H2

### DISCUSSION THREAD

**Discussion by shockogit** [genotrance/px#180] [Q&A]
I'm confused by the host only directive in the context of running on Windows 10 22 H2. When set to hostonly=1 it listens on 0.0.0.0 and hence accept a TCP connection on the port from a remote system. That said, it does not accept a HTTP connect so I assume this directive adds the local interface address to the allow list. Now the .ini file help states:

```
; Allow only local interfaces to use proxy. 0 or 1, default: 0
;   Px allows all IP addresses assigned to local interfaces to use the service.
;   This allows local apps as well as VM or container apps to use Px when in a
;   NAT config. Px does this by listening on all interfaces and overriding the
;   allow list.

```

Now I have tired this with both Hyper-V VMs behind the default Hyper-V switch and WSL but neither work. The reason seems to be that in the `hostonly=1` configuration PX proxy allow list is the virtual interface IP address for the Hyper-V and WSL instances but the VM/WSL traffic does not come to PX proxy via this address it comes via the WSL or VM instance IP address. My only solution  has been to create individual instances of PX proxy listening on the individual IP address of these virtual adapters.

**Comment by genotrance** [comment]
[v0.9.0](https://github.com/genotrance/px/releases/tag/v0.9.0) now supports multiple `listen` values so you no longer need to run multiple instances. I also have WSL running now so will look into improving support in the future.
